ruma-identity-service-api/src/lookup.rs

```diff
@@ -0,0 +1,49 @@
+//! Endpoints to look up Matrix IDs bound to 3PIDs.
+
+use ruma_serde_macros::{AsRefStr, DisplayAsRefStr, FromString};
+use serde::{Deserialize, Serialize};
+
+pub mod lookup_3pid;
+
+/// The algorithms that can be used to hash the identifiers used for lookup, as defined in the
+/// Matrix Spec.
+///
+/// This type can hold an arbitrary string. To check for algorithms that are not available as a
+/// documented variant here, use its string representation, obtained through `.as_str()`.
+#[derive(
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    AsRefStr,
+    DisplayAsRefStr,
+    FromString,
+    Serialize,
+    Deserialize,
+)]
+#[non_exhaustive]
+#[ruma_enum(rename_all = "snake_case")]
+#[cfg_attr(feature = "serde", derive(DeserializeFromCowStr, SerializeAsRefStr))]
```

Comment:
I guess `crypto_algorithms` was not the best reference. There is no need to use all these individual derives, there is one that does all of them. The `cfg_attr` is also not needed since `serde` is not an optional depdendency here. You can just use the [`ruma_serde::StringEnum`](https://docs.ruma.io/ruma/serde/derive.StringEnum.html) derive macro, remove the other ones and also remove the (direct) dependency on ruma-serde-macros.